src/plots/cartesian/select.js

```diff
@@ -190,6 +195,7 @@ module.exports = function prepSelect(e, startX, startY, dragOptions, mode) {
                 selection = [];
                 for(i = 0; i < searchTraces.length; i++) {
                     searchInfo = searchTraces[i];
+                    if(!visible(searchInfo)) continue;
```

Comment:
Nice. Does that mean we can :hocho: those `visible !== false` clauses [here](https://github.com/plotly/plotly.js/blob/master/src/traces/bar/select.js#L20), [here](https://github.com/plotly/plotly.js/blob/master/src/traces/scatter/select.js#L29), [here](https://github.com/plotly/plotly.js/blob/master/src/traces/scattergeo/select.js#L25), [here](https://github.com/plotly/plotly.js/blob/86e7135f54fedfa1e89b3e299833491369baa96e/src/traces/scattermapbox/select.js#L27) and [here](https://github.com/plotly/plotly.js/blob/master/src/traces/scattergl/select.js#L29)?

Reply:
Great find, indeed these are the only callers, and `select` check is almost always on the top, except `scattermapbox/select/js` where it's okay to set the `_hasDimmedPts` to `false` anyway:

```
    trace._hasDimmedPts = false;

    if(!subtypes.hasMarkers(trace)) return [];
```
Pushing a new commit now